### DISCUSSION TITLE
Why is max_join_length set to 120?

### DISCUSSION THREAD

**Discussion by jay-babu** [Wansmer/treesj#117] [Q&A]
Why is max_join_length set to 120? Can the reason why be documented somewhere?

**Comment by Wansmer** [comment]
The length limit for `join` is needed to prevent unwanted joining when the plugin is not called at the right place (e.g., to avoid joining the whole json root node).
The reason that it is 120 is that it seemed right to me, since there is usually no need to collapse/uncollapse longer nodes.
To disable this behavior, you can simply set this option to any very large number, such as 9999.

**Reply by jay-babu** [reply]
Would a default value of 180 or 240 be better? I think I hit 120 limit pretty often. https://github.com/search?q=max_join_length++language%3ALua&type=code&l=Lua&p=2. seems people override to some higher limit pretty often: https://github.com/search?q=%22Wansmer%2Ftreesj%22+language%3ALua&type=code&l=Lua

**Reply by Wansmer** [reply]
You may be right. I will think about changing this value in the future.

**Reply by cbochs** [reply]
Please don't change the default of 120.
